src/server.rs

```diff
@@ -732,3 +770,343 @@ async fn handle_hexists(
 
     Ok(())
 }
+
+pub fn compress(
+    data: &[u8],
+    config: &CompressionConfig,
+) -> Result<Vec<u8>, Box<dyn std::error::Error>> {
+    if !config.enabled || matches!(config.algorithm, CompressionType::None) {
+        return Ok(data.to_vec());
+    }
+
+    match config.algorithm {
+        CompressionType::None => Ok(data.to_vec()),
+        CompressionType::Gzip => {
+            use flate2::Compression;
+            use flate2::write::GzEncoder;
+
+            let level = config.level.unwrap_or(6);
+            let compression = Compression::new(level);
```

Comment:
Compression Initialization should not happen on each request.

Reply:
Compression Initialization is happening for each reqest, can add it to struct in gzip.rs impl